lib/ProductOpener/APIProductWrite.pm

```diff
@@ -85,6 +85,34 @@ sub skip_protected_field ($product_ref, $field, $moderator = 0) {
 	return 0;
 }
 
+=head2 skip_protected_image($product_ref, $imagefield, $moderator = 0)
+
+=cut
+
+sub skip_protected_image ($product_ref, $imagefield, $moderator = 0) {
+
+	# Add image-specific checks here
+	if (defined $product_ref->{imgupload_ $imagefield}) {
```

Comment:
I don't think this will work: `$product_ref->{imgupload_ $imagefield})`

For fields, we have the owner_fields structure that tells us if a field value was set by its owner.

For images, it's more complex, as currently at least, we don't have images in owner_fields

If you look at this product for instance, you can look at the "images" field:

https://world.openfoodfacts.org/api/v3/product/3596710384396/lentilles-corail-l-oiseau?fields=images

then if we check what we have for the "front_fr" image:

```
front_fr: {
angle: 0,
coordinates_image_size: "400",
geometry: "0x0--7--7",
imgid: "16",
normalize: "false",
rev: "57",
sizes: {
100: {
h: 100,
w: 66
},
200: {
h: 200,
w: 132
},
400: {
h: 400,
w: 263
},
full: {
h: 2248,
w: 1480
}
},
white_magic: "false",
x1: "-1",
x2: "-1",
y1: "-1",
y2: "-1"
},
```

we have "imgid: 16", so we can check who uploaded that image:

```
{
sizes: {
100: {
h: 100,
w: 86
},
400: {
h: 400,
w: 344
},
full: {
h: 2831,
w: 2438
}
},
uploaded_t: 1688498812,
uploader: "org-auchan-apaw"
},
```

and here we see it's the producer who uploaded the image:  (org-auchan-apaw)

So in skip_protected_image, we should do that: identify the imgid of the selected image, and then identify the uploader and check if the uploader is the owner of the product.

To simplify testing, I think it would be useful to first create a function get_selected_image_uploader("front_fr") that would return the uploader of a selected image, and have unit tests for it.